src/main/java/com/fundynamic/d2tm/game/map/renderer/MapCellViewportRenderer.java

```diff
@@ -26,17 +27,17 @@ public MapCellViewportRenderer(Map map, int tileHeight, int tileWidth, Vector2D
         cellsThatFitVertically = (windowDimensions.getY() / tileHeight) + 1;
     }
 
-    public void render(Image imageToDrawOn, Vector2D viewingVector, Renderer<MapCell> renderer) throws SlickException {
-        int startCellX = (viewingVector.getX() / tileWidth);
-        int startCellY = (viewingVector.getY() / tileHeight);
+    public void render(Image imageToDrawOn, Vector2f viewingVector, Renderer<MapCell> renderer) throws SlickException {
+        int startCellX = (int)(viewingVector.getX() / tileWidth);
```

Comment:
If we change `Vector2D` to be more floaty, it becomes even more like `Vector2f`. I don't see why we should implement a custom type when we can use one that is built-in. Besides, some usages of `Vector2D` now depend on in being an integer...